src/nomad_simulations/schema_packages/basis_set.py

```diff
@@ -190,16 +190,82 @@ class AtomCenteredFunction(ArchiveSection):
     Specifies a single function (term) in an atom-centered basis set.
     """
 
-    pass
+    function_type = Quantity(
+        type=MEnum('S', 'P', 'D', 'F', 'G', 'H', 'I', 'J'),
+        description="""
+        the angular momentum of the shell to be added.
+        """,
+    )
+
+    n_primitive = Quantity(
+        type=int,
+        description="""
+        Number of primitives.
+        """,
+    )
+
+    exponents = Quantity(
+        type=np.float32,
+        shape=['n_primitive'],
+        description="""
+        List of exponents for the basis function.
+        """,
+    )
 
-    # TODO: design system for writing basis functions like gaussian or slater orbitals
+    contraction_coefficients = Quantity(
+        type=np.float32,
+        shape=['n_primitive'],
+        description="""
+        List of contraction coefficients corresponding to the exponents.
+        """,
+    )
+
+    def normalize(self, archive: 'EntryArchive', logger: 'BoundLogger') -> None:
+        super().normalize(archive, logger)
+        # self.name = self.m_def.name
 
 
 class AtomCenteredBasisSet(BasisSetComponent):
     """
     Defines an atom-centered basis set.
     """
 
+    main_basis_set = Quantity(
+        type=str,
+        description="""
+        Name of the main basis set.
+        """,
+    )
```

Comment:
What's the distinction between `main` and `aux`? If it's method routines, you can use `BasisSetContainer` to distinguish them.